test/algorithms.jl

```diff
@@ -0,0 +1,28 @@
+
+@testset "replace! & replace" begin
```

Comment:
new test files would need to be added to test/choosetests.jl to actually run, but aren't there existing files with related tests this would fit with?